common/src/main/java/com/wynntils/wynn/model/questbook/QuestInfo.java

```diff
@@ -95,6 +103,50 @@ public String toString() {
                 + additionalRequirements + "]";
     }
 
+    public static List<Component> getTooltipLinesForQuest(QuestInfo questInfo) {
+        List<Component> tooltipLines = new ArrayList<>();
+
+        tooltipLines.add(new TextComponent(questInfo.getName())
+                .withStyle(ChatFormatting.BOLD)
+                .withStyle(ChatFormatting.WHITE));
+        tooltipLines.add(questInfo.getStatus().getQuestBookComponent());
+        tooltipLines.add(new TextComponent(""));
+        tooltipLines.add((CharacterManager.getCharacterInfo().getLevel() >= questInfo.getLevel()
+                        ? new TextComponent("✔").withStyle(ChatFormatting.GREEN)
+                        : new TextComponent("✖").withStyle(ChatFormatting.RED))
+                .append(new TextComponent(" Combat Lv. Min: ").withStyle(ChatFormatting.GRAY))
+                .append(new TextComponent(String.valueOf(questInfo.getLevel())).withStyle(ChatFormatting.WHITE)));
+
+        for (Pair<String, Integer> additionalRequirement : questInfo.getAdditionalRequirements()) {
+            MutableComponent base = CharacterManager.getCharacterInfo()
+                                    .getProfessionInfo()
+                                    .getLevel(ProfessionInfo.ProfessionType.valueOf(additionalRequirement.a))
+                            >= additionalRequirement.b
+                    ? new TextComponent("✔ ").withStyle(ChatFormatting.GREEN)
+                    : new TextComponent("✖ ").withStyle(ChatFormatting.RED);
+
+            tooltipLines.add(base.append(new TextComponent(additionalRequirement.a + " Lv. Min: ")
+                    .withStyle(ChatFormatting.GRAY)
+                    .append(new TextComponent(String.valueOf(additionalRequirement.b))
+                            .withStyle(ChatFormatting.WHITE))));
+        }
+
+        tooltipLines.add(new TextComponent("-")
+                .withStyle(ChatFormatting.GREEN)
+                .append(new TextComponent(" Length: ").withStyle(ChatFormatting.GRAY))
+                .append(new TextComponent(StringUtils.capitalizeFirst(
+                                questInfo.getLength().toString().toLowerCase(Locale.ROOT)))
+                        .withStyle(ChatFormatting.WHITE)));
+        tooltipLines.add(new TextComponent(""));
```

Comment:
Maybe this is nitpicking, but I think it would be nicer if this part ended without the trailing empty line, and that this instead was added as the first thing by the questbook screen gui.